packages/inventory/src/EntityTableToolbar.js

```diff
@@ -36,60 +42,107 @@ class ContextEntityTableToolbar extends Component {
         });
     }
 
-    onSetTextFilter = (value) => {
-        const { page, perPage, filters } = this.props;
+    onSetTextFilter = (value, debounced = true) => {
+        const { onRefreshData, perPage, filters } = this.props;
         const textualFilter = filters.find(oneFilter => oneFilter.value === TEXT_FILTER);
         if (textualFilter) {
             textualFilter.filter = value;
         } else {
             filters.push({ value: TEXT_FILTER, filter: value });
         }
 
-        this.debouncedRefresh({ page: 1, perPage, filters });
-        this.setState({ textFilter: value });
+        const refresh = debounced ? this.debouncedRefresh : onRefreshData;
+        this.setState({ textFilter: value }, () => refresh({ page: 1, perPage, filters }));
     }
 
     applyTags = (newSelection, debounced = true) => {
-        const { allTags, page, perPage, filters, onRefreshData } = this.props;
-        const tagFilters = mapGroups(newSelection, allTags);
+        const { perPage, filters, onRefreshData } = this.props;
+        const tagFilters = mapGroups(newSelection);
         const tagFiltersIndex = filters.findIndex((value) => value.hasOwnProperty('tagFilters'));
-        filters.splice(tagFiltersIndex, 1);
+        if (tagFiltersIndex !== -1) {
```

Comment:
I think it is not necessary to use `splice` here. Something like `filters.filter(value) => !value.hasOwnProperty('tagFilters'))` should be enough.

Reply:
Yeah, you are right!